src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -226,6 +226,7 @@ private Task RestartAsync()
         {
             if (!_shutdownScheduled)
             {
+                _eventManager.Publish(new WorkerShutdownEvent());
```

Comment:
Note: we should change this to directly reference the language worker channel manager to make sure that restart happens.